subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java

```diff
@@ -319,7 +320,11 @@ protected ProjectLayout getProjectLayout() {
     @Nested
     @Deprecated
     protected JavaPlatform getPlatform() {
-        return new DefaultJavaPlatform(JavaVersion.toVersion(getTargetCompatibility()));
+        Object targetCompatibility = getTargetCompatibility();
```

Comment:
I do not get this change. Tbh I do not get the code that was there before either 😉 
Can you explain what this fixes / improves?

Reply:
As discussed, this was only needed under the previous assumption of supporting non-toolchain compiler daemons. Reverted.